lib/metka/model.rb

```diff
@@ -1,47 +1,79 @@
 # frozen_string_literal: true
 
+require 'arel'
+
 module Metka
-  def self.Model(column:, **options)
-    Metka::Model.new(column: column, **options)
+  OR = Arel::Nodes::Or
+  AND = Arel::Nodes::And
+
+  def self.Model(column: nil, columns: nil, **options)
+    columns = [column, *columns].uniq.compact
+    raise ArgumentError, 'Columns not specified' unless columns.present?
+
+    Metka::Model.new(column: column, columns: columns, **options)
   end
 
   class Model < Module
-    def initialize(column: , **options)
-      @column = column
-      @options = options
+    def initialize(columns: nil, **options)
+      @columns = columns.dup.freeze
+      @options = options.dup.freeze
     end
 
     def included(base)
-      column = @column
+      columns = @columns
       parser = ->(tags) {
         @options[:parser] ? @options[:parser].call(tags) : Metka.config.parser.instance.call(tags)
       }
 
       search_by_tags = ->(model, tags, column, **options) {
         parsed_tag_list = parser.call(tags)
-        if options[:without].present?
-          model.where.not(::Metka::QueryBuilder.new.call(model, column, parsed_tag_list, options))
-        else
-          return model.none if parsed_tag_list.empty?
-          model.where(::Metka::QueryBuilder.new.call(model, column, parsed_tag_list, options))
-        end
+        return model.none if parsed_tag_list.empty?
+
+        model.where(::Metka::QueryBuilder.new.call(model, column, parsed_tag_list, options))
+      }
+
+      # @param model [ActiveRecord::Base] model on which to execute search
+      # @param tags [Object] list of tags, representation depends on parser used
+      # @param options [Hash] options
+      #   @option :join_operator [Metka::AND, Metka::OR]
+      # @returns ViewPost::ActiveRecord_Relation
+      tagged_with_lambda = ->(model, tags, **options) {
+        parsed_tag_list = parser.call(tags)
+
+        return model.none if parsed_tag_list.empty?
+
+        request = ::Metka::QueryBuilder.new.call(model, columns, parsed_tag_list, options)
+        model.where(request)
       }
```

Comment:
Maybe those two `search_by_tags` and `tagged_with_lambda` can be united? It looks like the only real difference is that the `tagged_with_lambda` doesn't require `column` attribute to be passed, so it is basically `search_by_tags` with `@columns` passed as `column` attribute's value?

Reply:
Yeah, not quite, but yes :) It was almost necessary to have it until we decided to make breaking changes and to specify all tag columns at once.